libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -329,6 +335,20 @@ def add_texts(
         """Add texts data to an existing index."""
         keys = kwargs.get("keys")
         ids = []
+
+        # batching support if embedding function is an Embeddings object
+        if isinstance(self.embedding_function, Embeddings):
+            try:
+                embeddings = self.embedding_func.embed_documents(texts)
```

Comment:
should this be `embedding_function`? Same below

Reply:
fixed.